File: src/main/java/pl/training/workshop/m6/s07_decorator/step3/Insurance.java
package pl.training.workshop.m6.s07_decorator.step3;

import pl.training.workshop.shared.Money;

/**
 * Krok 3: bez zmian - ubezpieczenie jest zawsze najbardziej zewnętrzne,
 * bo w opisie występuje na końcu.
 */
public record Insurance(PricedTicket inner) implements PricedTicket {
    @Override
    public Money price() {
        return inner.price().plus(Money.of("4.00"));
    }

    @Override
    public String description() {
        return inner.description() + " +ubezpieczenie";
    }
}

File: csharp/src/Training.Workshop/M6/S07Decorator/Step3/Insurance.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M6.S07Decorator.Step3;

/// <summary>
/// Krok 3: bez zmian - ubezpieczenie jest zawsze najbardziej zewnętrzne,
/// bo w opisie występuje na końcu.
/// </summary>
public sealed record Insurance(IPricedTicket Inner) : IPricedTicket
{
    public Money Price()
    {
        return Inner.Price().Plus(Money.Of("4.00"));
    }

    public string Description()
    {
        return Inner.Description() + " +ubezpieczenie";
    }
}
